metrics/web/lib/project_groups/presentation/state/project_groups_notifier.dart

```diff
@@ -0,0 +1,329 @@
+  void toggleProjectCheckedStatus({String projectId, bool isChecked}) {
+    final projectIds = _activeProjectGroupDialogViewModel.selectedProjectIds;
+
+    if (isChecked) {
+      projectIds.add(projectId);
+    } else {
+      projectIds.remove(projectId);
+    }
+
+    final projectIndex = _projectSelectorViewModels
+        .indexWhere((project) => project.id == projectId);
+
+    final project = _projectSelectorViewModels[projectIndex];
+
+    _projectSelectorViewModels[projectIndex] = ProjectSelectorViewModel(
+      id: project.id,
+      name: project.name,
+      isChecked: isChecked,
+    );
+
+    _activeProjectGroupDialogViewModel = ActiveProjectGroupDialogViewModel(
+      id: _activeProjectGroupDialogViewModel.id,
+      name: _activeProjectGroupDialogViewModel.name,
+      selectedProjectIds: projectIds,
+    );
+
+    notifyListeners();
+  }
+
+  /// Subscribes to project groups.
+  Future<void> subscribeToProjectGroups() async {
+    final projectGroupsStream = _receiveProjectGroupUpdates();
+    _errorMessage = null;
+    await _projectGroupsSubscription?.cancel();
+    _projectGroupsSubscription = projectGroupsStream.listen(
+      _projectGroupListener,
+      onError: _errorHandler,
+    );
+  }
+
+  /// Unsubscribes from project groups.
+  Future<void> unsubscribeFromProjectGroups() async {
+    await _cancelSubscriptions();
+    notifyListeners();
+  }
+
+  /// Saves project group data into Firestore with the given [projectGroupName],
+  /// [projectIds].
+  ///
+  /// If [projectIds] is null, a new project group is added,
+  /// otherwise existing ones are updated.
+  Future<bool> saveProjectGroup(
+    String projectGroupId,
+    String projectGroupName,
+    List<String> projectIds,
+  ) async {
+    resetFirestoreWriteErrorMessage();
+
+    try {
+      if (projectGroupId == null) {
+        await _addProjectGroupUseCase(
+          AddProjectGroupParam(
+            projectGroupName: projectGroupName,
+            projectIds: projectIds,
+          ),
+        );
+      } else {
+        await _updateProjectGroupUseCase(
+          UpdateProjectGroupParam(
+            projectGroupId,
+            projectGroupName,
+            projectIds,
+          ),
+        );
+      }
+    } catch (e) {
+      _firestoreWriteErrorHandler(e);
+    }
+
+    return _firestoreWriteErrorMessage == null;
+  }
+
+  /// Deletes project group data from Firestore with the given [projectGroupId].
+  Future<bool> deleteProjectGroup(String projectGroupId) async {
+    resetFirestoreWriteErrorMessage();
+
+    try {
+      await _deleteProjectGroupUseCase(
+        DeleteProjectGroupParam(projectGroupId: projectGroupId),
+      );
+    } catch (e) {
+      _firestoreWriteErrorHandler(e);
+    }
+
+    return _firestoreWriteErrorMessage == null;
+  }
+
+  /// Sets [_firestoreWriteErrorMessage] to null.
+  void resetFirestoreWriteErrorMessage() {
+    _firestoreWriteErrorMessage = null;
+    notifyListeners();
+  }
```

Comment:
```suggestion
  void resetProjectGroupSavingError() {
    _projectGroupSavingError = null;
    notifyListeners();
  }
```

Also, maybe this method should be private?